docs/howto/krustlet-on-kubernetes-with-inlets.md

```diff
@@ -0,0 +1,273 @@
+```
+
+## Step 1: Create a service account user for the node
+
+We will need to create a Kubernetes configuration file (known as the kubeconfig) and service account
+for krustlet to use to register nodes and access specific secrets.
+
+This can be done by using the Kubernetes manifest in the [assets](./assets) directory:
+
+```shell
+$ kubectl apply -n kube-system -f ./docs/howto/assets/krustlet-service-account.yaml
+```
+
+You can also do this by using the manifest straight from GitHub:
+
+```shell
+$ kubectl apply -n kube-system -f https://raw.githubusercontent.com/deislabs/krustlet/master/docs/howto/assets/krustlet-service-account.yaml
+```
+
+Now that things are all set up, we need to generate the kubeconfig. You can do this by running
+(assuming you are in the root of the krustlet repo):
+
+```shell
+$ ./docs/howto/assets/generate-kubeconfig.sh
+```
+
+Or if you are feeling a bit more trusting, you can run it straight from the repo:
+
+```shell
+bash <(curl https://raw.githubusercontent.com/deislabs/krustlet/master/docs/howto/assets/generate-kubeconfig.sh)
+```
+
+Either way, it will output a file called `kubeconfig-sa` in your current directory. Save this for
+later.
+
+## Step 2: Create Certificate
+
+Krustlet requires a certificate for securing communication with the Kubernetes API. Because
+Kubernetes has its own certificates, we'll need to get a signed certificate from the Kubernetes API
+that we can use. First things first, let's create a certificate signing request (CSR):
+
+```shell
+$ openssl req -new -sha256 -newkey rsa:2048 -keyout krustlet.key -out krustlet.csr -nodes -subj "/C=US/ST=./L=./O=./OU=./CN=krustlet"
+```
+
+This will create a CSR and a new key for the certificate, using `krustlet` as the hostname of the
+server.
+
+Now that it is created, we'll need to send the request to Kubernetes:
+
+```shell
+$ cat <<EOF | kubectl apply -f -
+apiVersion: certificates.k8s.io/v1beta1
+kind: CertificateSigningRequest
+metadata:
+  name: krustlet
+spec:
+  request: $(cat krustlet.csr | base64 | tr -d '\n')
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+EOF
+certificatesigningrequest.certificates.k8s.io/krustlet created
+```
+
+Once that runs, an admin (that is probably you! at least it should be if you are trying to add a
+node to the cluster) needs to approve the request:
+
+```shell
+$ kubectl certificate approve krustlet
+certificatesigningrequest.certificates.k8s.io/krustlet approved
+```
+
+After approval, you can download the cert like so:
+
+```shell
+$ kubectl get csr krustlet -o jsonpath='{.status.certificate}' \
+    | base64 --decode > krustlet.crt
+```
+
+Lastly, combine the key and the cert into a PFX bundle, choosing your own password instead of
+"password":
+
+```shell
+$ openssl pkcs12 -export -out krustlet.pfx -inkey krustlet.key -in krustlet.crt -password "pass:password"
+```
+
+## Step 3: Setup the inlets server and client
+
+* Create a Kubernetes secret for inlets
```

Comment:
To remain consistent with the rest of the documentation, can you remove the bullet points?

If you want to break these into sub-sections, consider using sub-headers (`### Create a Kubernetes Secret`).

Reply:
😁

Reply:
😁